streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -296,9 +297,10 @@ private boolean hasPartitionsForAnyTopics(final List<String> topicNames, final S
     }
 
     private Set<String> getStoresOnHost(final Map<String, List<String>> storeToSourceTopics,
-        final Set<TopicPartition> sourceTopicPartitions, final String topologyName) {
+                                        final Set<TopicPartition> sourceTopicPartitions,
+                                        final String topologyName) {
         final InternalTopologyBuilder builder = topologyMetadata.lookupBuilderForNamedTopology(topologyName);
-        final Set<String> sourceTopicNames = builder.sourceTopicNames();
+        final Collection<String> sourceTopicNames = builder.sourceTopicCollection();
```

Comment:
Do we want the raw topic names (without the prefix) or the decorated ones here? 

BTW The function/variable names are a bit confusing but they stored different things. Maybe we should just rename them to be more clear.

Reply:
We want the actual ("decorated") topic names. PR #11562 introduced a bug by using the `sourceTopicNames` with only the raw topic names, this is one of the main IQ fixes. 

100% agree that the names are confusing, I was going to address that in #11601 but I suppose it makes sense to do that in this PR as well

Reply:
Ack! Thanks.